src/MockBot.Api/Controllers/ChatController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using MockBot.Api.Interfaces;
+using MockBot.Api.Models;
+
+namespace MockBot.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ChatController : ControllerBase
+    {
+        private readonly IChatService _chatService;
+
+        public ChatController(IChatService chatService)
+        {
+            _chatService = chatService;
+        }
+
+        [HttpGet]
+        public IEnumerable<Chat> FindAll()
+        {
+            return _chatService.FindAll();
+        }
+
+        [HttpGet("{id:guid}")]
+        public Chat Get(Guid id)
+        {
+            return _chatService.GetId(id);
+        }
+
+        [HttpPost]
+        public Chat Post()
+        {
+            return _chatService.CreateChat();
+        }
+
+        [HttpGet("{chatId:guid}/messages")]
+        public IEnumerable<Message> GetMessages(Guid chatId)
```

Comment:
Not sure this feels quite right.... I feels like there should be a separate `Messages` controller but lets discuss it

Reply:
Yes, this is a bit lazy, but I was trying to keep it as simple as possible. I can change it if you'd like.